app/javascript/packs/application.tsx

```diff
@@ -84,7 +86,14 @@ initReact({
       status={data.graph.status}
     />
   ),
-  'student-editor': (data: any) => <Student.Editor endpoint={data.endpoint} />,
+  'student-editor': (data: any) => (
+    <Student.Editor
+      endpoint={data.endpoint}
+      initialSubmission={
+        (camelizeKeys(data.submission) as unknown) as Submission
```

Comment:
I'm unsure whether where this should live. I think it's okay here.

Reply:
Agreed. It would be weird to do that in the editor component.